pulsar-broker/src/main/java/org/apache/pulsar/broker/service/nonpersistent/NonPersistentTopic.java

```diff
@@ -149,9 +149,9 @@ public NonPersistentTopic(String topic, BrokerService brokerService) {
                         isEncryptionRequired = false;
                     } else {
                         Policies policies = optPolicies.get();
+                        updateTopicPolicyByNamespacePolicy(policies);
                         isEncryptionRequired = policies.encryption_required;
                         isAllowAutoUpdateSchema = policies.is_allow_auto_update_schema;
-                        topicPolicies.getInactiveTopicPolicies().updateNamespaceValue(policies.inactive_topic_policies);
                         setSchemaCompatibilityStrategy(policies);
                         schemaValidationEnforced = policies.schema_validation_enforced;
```

Comment:
In the past, only InactiveTopicPolicies was updated, but now 3 types of Policies will be updated. Is it necessary? Is this modification related to this PR?

Reply:
> Is it necessary? Is this modification related to this PR?

Not necessary actually. The other two topic policies are not used in non-persistent topic, yet.
Doing this just want to make the topic policy update procedure consistent, as we are not distinguishing PersistentTopicPolicy and NonPersistentTopicPolicy. Some policy maybe just not implemented in NonPersistentTopic yet.